timelock-agent/src/main/java/com/palantir/atlasdb/timelock/paxos/ClientPaxosResourceFactory.java

```diff
@@ -117,19 +124,154 @@ private static NetworkClientFactories factories(
+            return UUID.randomUUID();
+        }
 
         @Value.Derived
-        default List<Closeable> closeables() {
-            return networkClientFactories().closeables();
+        default int quorumSize() {
+            return PaxosRemotingUtils.getQuorumSize(clusterAddress());
+        }
+
+        @Value.Derived
+        default Set<String> clusterAddress() {
+            return PaxosRemotingUtils.getClusterAddresses(install());
+        }
+
+        @Value.Derived
+        default Set<String> remoteUris() {
+            return PaxosRemotingUtils.getRemoteServerPaths(install());
+        }
+
+        @Value.Derived
+        default Path dataDirectory() {
+            return install().paxos().dataDirectory().toPath();
+        }
+
+        @Value.Derived
+        default Optional<TrustContext> trustContext() {
+            return PaxosRemotingUtils
+                    .getSslConfigurationOptional(install())
+                    .map(SslSocketFactories::createTrustContext);
+        }
+
+    }
+
+    @Value.Immutable
+    public abstract static class PaxosRemoteClients {
+
+        @Value.Parameter
+        public abstract TimelockPaxosInstallationContext context();
+
+        @Value.Parameter
+        public abstract TaggedMetricRegistry metrics();
+
+        @Value.Derived
+        public List<TimelockPaxosAcceptorRpcClient> nonBatchAcceptor() {
+            return createInstrumentedRemoteProxies(TimelockPaxosAcceptorRpcClient.class, "paxos-acceptor-rpc-client");
+        }
+
+        @Value.Derived
+        public List<TimelockPaxosLearnerRpcClient> nonBatchLearner() {
+            return createInstrumentedRemoteProxies(TimelockPaxosLearnerRpcClient.class, "paxos-learner-rpc-client");
+        }
+
+        @Value.Derived
+        public List<BatchPaxosAcceptorRpcClient> batchAcceptor() {
+            return createInstrumentedRemoteProxies(
+                    BatchPaxosAcceptorRpcClient.class,
+                    "batch-paxos-acceptor-rpc-client");
+        }
+
+        @Value.Derived
+        public List<BatchPaxosLearnerRpcClient> batchLearner() {
+            return createInstrumentedRemoteProxies(
+                    BatchPaxosLearnerRpcClient.class,
+                    "batch-paxos-learner-rpc-client");
+        }
+
+        private <T> List<T> createInstrumentedRemoteProxies(Class<T> clazz, String name) {
+            return context().remoteUris().stream()
+                    .map(uri -> AtlasDbHttpClients.DEFAULT_TARGET_FACTORY.createProxy(
+                            context().trustContext(),
+                            uri,
+                            clazz,
+                            name,
+                            false))
+                    .map(proxy -> AtlasDbMetrics.instrumentWithTaggedMetrics(
+                            metrics(),
+                            clazz,
+                            proxy,
+                            name,
+                            _unused -> ImmutableMap.of()))
```

Comment:
I think normally if we don't provide tags we just use `AtlasDbMetrics.instrument(metrics(), clazz, proxy, name)` (unless there's some reason we need this to be the tagged metrics registry? I've somewhat lost track of how the instrumentation goes, honestly)

Reply:
I think at some point I had a tag that was in here, but it might have gotten lost. For the timelock metrics, the naming scheme is just plain text, whereas  if I use the other stuff I believe I'll end up using the class name which is overly verbose.

Reply:
`AtlasDbMetrics#instrument` is overloaded - there is a `String name` that is optional, but if provided will just use the plain text (if not provided uses the class name). Don't feel strongly, in any case